### DISCUSSION TITLE
What would be the correct way to type the first argument on virtual field resolve?

### DISCUSSION THREAD

**Discussion by julirofrano** [keystonejs/keystone#7207] [Questions] [answered]
In [this](https://github.com/keystonejs/keystone/blob/main/examples/virtual-field/schema.ts#L45) example it add an any type but how should be the correct way?

**Comment by dcousens** [comment, marked as answer]
Hi @julirofrano,

Please see https://github.com/keystonejs/keystone/pull/7396/files for the easiest solution to type your configuration and `resolve` functions.

```js
import { Lists } from '.keystone/types';

// ...

export const lists: Lists = {
// ...
```

If you needed to type this statically without infererence,  and the type should have been a `Post`,   you could have used the following:

```typescript
resolve: (item: Lists.Post.Item) => {
  // ...
```
